src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXToZHigh.cpp

```diff
@@ -126,28 +126,53 @@ Value getLSTMGRUGetYh(Location loc, PatternRewriter &rewriter, Value val,
                          loc, nullptr, rewriter.getI64TensorAttr({INT_MAX}))
                      .getResult();
   StringRef directionStr = direction.getValue();
-  Value start, end;
-  if (directionStr.equals_insensitive("forward")) {
-    start = minusOne;
-    end = intMax;
-  } else if (directionStr.equals_insensitive("reverse")) {
-    start = zero;
-    end = one;
-  } else {
-    llvm_unreachable("Bidirectional is not supported.");
-  }
-  ArrayRef<int64_t> yhShape =
+  ArrayRef<int64_t> resYhShape =
       resYh.getType().cast<RankedTensorType>().getShape();
-  SmallVector<int64_t> sliceShape({1, yhShape[0], yhShape[1], yhShape[2]});
-  Type elementType = resYh.getType().cast<RankedTensorType>().getElementType();
-  Type sliceType = RankedTensorType::get(sliceShape, elementType);
+  int64_t T = isNoneType(resY) ? 1 : shapeX[0];
+  int64_t D = resYhShape[0];
+  int64_t B = resYhShape[1];
+  int64_t H = resYhShape[2];
+  Type elementType = resYh.getType().cast<ShapedType>().getElementType();
   Value axis = zero;
   Value step = one;
-  ONNXSliceOp sliceOp =
-      rewriter.create<ONNXSliceOp>(loc, sliceType, val, start, end, axis, step);
-  ONNXSqueezeV11Op squeezeOp = rewriter.create<ONNXSqueezeV11Op>(
-      loc, resYh.getType(), sliceOp.getResult(), rewriter.getI64ArrayAttr(0));
-  return squeezeOp.getResult();
+  Value ret;
+  if (directionStr.equals_insensitive("forward") ||
```

Comment:
This string comparisons can be avoided by making the type of direction a C++11 `enum class`.  The code here can then be rewritten as: 

```
switch (direction) {
case Direction::reverse:
  ...
  break;
case Direction::forward:
   ... 
   break;
   etc...
}
```

Reply:
I am thinking to create another PR to use enum class for the direction values in zHigh and zlow directives as same to the above.

Reply:
Yes. I will use switch statement after using `enum class` for direction. Thank you.

Reply:
I now think we should do that in a future PR for the reasons I outlined in this [thread](https://github.com/onnx/onnx-mlir/pull/1268/files#r835268155).